### ISSUE TITLE
Upload Python Wheel to PyPI to support Windows and update README

### ISSUE THREAD

**Issue by CzarekTomczak** [cyaninc/git-fat#39]
I see that there was a new release 0.4.0 on PyPI, but there is still no Python Wheel package available. This means you can't install git-fat on Windows. It will install using the .tar.gz, but it won't work. A python wheel package is required that contains windows binaries. It can be generated by running the win32/setup_wheel.bat script.

Regarding README, I think the whole "Installation on Windows" section should be removed. It's confusing and not really needed. Maybe only just add the note that the Python/Scripts directory must be in PATH, in some other place in README.

Also it should be noted in README that currently you can't install it Windows using the pip command. It can be installed from sources by running the win32/setup_wheel.bat script.

@abraithwaite Let me know if you would like me to generate the Wheel package and send it to you. If you plan on doing it later on your own, then maybe just update README in the meantime to avoid confusion.

**Comment by justinclift** [comment]
Yeah, making the Wheel stuff available sounds like a good idea to help people on Windows move forward.  Update the README to reflect any Windows specific usage/requirements should be done at the same time too (just saying).

**Comment by abraithwaite** [comment]
Yes, it slipped my mind that there might be another step in the release process for Windows.  Since I don't have access to a windows machine, I would appreciate it if you could send the wheel package to me for upload.

**Comment by abraithwaite** [comment]
Yes, it slipped my mind that there might be an additional installation step for windows users.

However, looking deeper it appears that it should be possible to combine the windows/unix packages into one wheel package.  Unless you think that can't be done, I'm going to give it a shot.  I'll update the readme as well.  (I have never used wheel before but the site http://pythonwheels.com/ makes it sound easy if there aren't any c extensions).

**Comment by CzarekTomczak** [comment]
The Wheel package is a way to distribute package on PyPI that contains platform specific binaries. I don't think you can create a single wheel package for all platforms in our case. On Windows you need to ship Windows specific binaries, so a separate package tagged for Windows platform is the way to go. The generated wheel file is named like "git_fat-0.3.4-cp27-none-win32.whl". It contains Python tag (cp27) and platform tag (win32). I think it's okay to provide .tar.gz source package for Linux/Mac and .whl for Win. I've tested it and pip will prefer the .whl package when installing, given that the latest source package is the same version as the .whl package. If you upload a new version with only .tar.gz package, then pip will install that package on Windows and it won't work. So it's important to always upload the .whl package along with .tar.gz when doing a release.

**Comment by abraithwaite** [comment]
I think you're right. Looking at the win32 directory more closely, it seems like it might be difficult to be just a source distribution.

If you send me the 0.4.0 .whl package, I can manually add it this time around and next time we can figure out something more automated and less fragmented.

**Comment by CzarekTomczak** [comment]
I've uploaded .whl packages to Google Drive - download from [here](https://drive.google.com/folderview?id=0B1di2XiBBfacflNDbkxXc3pLUGJOLTB4azNjenJTU3pSQ1NDNUxuWmVmTDluMS1nbHJmZ00&usp=sharing#list).

**Comment by abraithwaite** [comment]
Uploaded

**Comment by CzarekTomczak** [comment]
@abraithwaite README still needs updating

**Comment by abraithwaite** [comment]
Yes, thank you.

**Comment by abraithwaite** [comment]
Hmm, so looking at doing another release, I was thinking about doing this:

>  next time we can figure out something more automated and less fragmented.

But then I ran into this (according to pythonwheels.org):

> You will need to have access to the platform you are building for.

If I release a new version, would you mind building a wheel package for windows?

**Comment by CzarekTomczak** [comment]
@abraithwaite Not a problem

**Comment by stemann** [comment]
So... what is needed to get git-fat running on Windows?

**Comment by abraithwaite** [comment]
@stemann I think it's installable as a `.whl` for windows now.  I'm still learning about how those work, but I don't have a windows machine to try it out.  Sorry!

**Comment by abraithwaite** [comment]
@CzarekTomczak I just did another release, so if you would be so kind as to build another windows package, I think many would appreciate it. :shipit:

As an aside, know of any free for OSS windows build servers?

**Comment by stemann** [comment]
@abraithwaite Thanks, it worked out.

You could run the Windows build on an on-demand Windows Vagrant box:
https://atlas.hashicorp.com/boxes/search?utf8=✓&sort=&provider=&q=windows

**Comment by cztomczak** [comment]
@abraithwaite Uploaded, see [here](https://drive.google.com/folderview?id=0B1di2XiBBfacflNDbkxXc3pLUGJOLTB4azNjenJTU3pSQ1NDNUxuWmVmTDluMS1nbHJmZ00&usp=sharing#list). Tests run fine.
